drivers/source/serial_io.c

```diff
@@ -64,6 +67,16 @@ obc_error_code_t sciPrintText(unsigned char *text, uint32_t length) {
 
     /* USER CODE BEGIN */
     // Print text to the serial port using sciSendBytes. Use the mutex to protect the SCI module.
+    if (xSemaphoreTake(mutex, UART_MUTEX_BLOCK_TIME) == pdTRUE) {
+
+        obc_error_code_t result = sciSendBytes(text, length);
+        if (xSemaphoreGive(mutex) == pdTRUE)
+            return OBC_ERR_CODE_QUEUE_FULL;
+
+        return result;
+    } else {
+        return OBC_ERR_CODE_MUTEX_TIMEOUT;
+    }
```

Comment:
No need for the `else` branch since the `if ` branch returns. You can just do:
```
if (...) {
  ...
  return result;
}

return OBC_ERR_CODE_MUTEX_TIMEOUT;
```